### PR TITLE
fungible token create, mint, burn API added

### PR DESCRIPTION
**Pull request by rex-god**

### CODE REVIEW

File: src/modules/token/components/burn-token/burn-token.controller.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@ApiTags('Token')
+@ApiSecurity('api_key', ['x-api-key'])
+@Controller('delete')
+export class BurnTokenController {
+  constructor(private burnTokenService: BurnTokenService) {}
+
+  @BurnTokenOpenApi()
+  @Delete('burn')
+  @Version('1')
+  async createToken(@Body() burnTokenDto: BurnTokenDto): Promise<any> {
```
Comment (codinpsycho) [inline src/modules/token/components/burn-token/burn-token.controller.ts:16]: Function name?

File: src/modules/token/components/burn-token/burn-token.service.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        feePayer,
+        tokenAccountOwner[0],
+        tokenAddressPubkey,
+        feePayer,
+        LAMPORTS_PER_SOL * amount,
+        9,
+      );
+      return txhash;
+    } catch (err) {
+      console.log(err);
```
Comment (codinpsycho) [inline src/modules/token/components/burn-token/burn-token.service.ts:48]: throw error as well?

File: src/modules/token/components/burn-token/open-api.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  return applyDecorators(
+    ApiOperation({ summary: 'Burn Token' }),
+    ApiOkResponse({
+      description: 'Token burned successfully',
+      schema: {
+        example: {
+          success: true,
+          message: 'Token burned successfully',
+          result: {
+            txId: '4qUvyoFd7dfbsdRWiXaTV9zdpCJS7ZAzXGQQET1cFcbaXJ1f539MnDbmKaGGxKDbaFjyJjSJ6UvDk5ytRPqfSPAb',
```
Comment (codinpsycho) [inline src/modules/token/components/burn-token/open-api.ts:15]: Are these the parameters in result for burn?

File: src/modules/token/components/create-token/create-token.service.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      ),
+    );
+    const txhash = await actions.sendTransaction({
+      connection,
+      wallet,
+      txs: [createNewTokenTransaction],
+      signers: [mintKeypair],
+    });
+
+    return { txhash, mint_token_address: mintKeypair.publicKey.toBase58() };
```
Comment (codinpsycho) [inline src/modules/token/components/create-token/create-token.service.ts:103]: Only token_address is sufficient as name?

File: src/modules/token/components/create-token/create-token.service.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+    const lamports = await getMinimumBalanceForRentExemptMint(connection);
+    const mintKeypair = Keypair.generate();
+    const metadataPDA = findMetadataPda(mintKeypair.publicKey);
+    const tokenATA = await getAssociatedTokenAddress(
+      mintKeypair.publicKey,
+      feePayer.publicKey,
+    );
+
+    const tokenMetadata = {
```
Comment (codinpsycho) [inline src/modules/token/components/create-token/create-token.service.ts:41]: For fungible we don't need royalty and share, lets keep it as 0 and 100 by default? No need to get from user.

File: src/modules/token/components/mint-token/dto/mint-token.dto.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    type: Number,
+    description: 'How many tokens do you want to mint?',
+    example: 10,
+  })
+  @IsNotEmpty()
+  @IsNumber()
+  readonly amount: number;
+
+  @ApiProperty({
+    title: 'Decimals',
```
Comment (codinpsycho) [inline src/modules/token/components/mint-token/dto/mint-token.dto.ts:46]: Are decimals required as a parameter from user?

Comment (rex-god) [reply src/modules/token/components/mint-token/dto/mint-token.dto.ts:46]: Don't know the actual concept of it. I can be changeable that's why I've put as user input.

File: src/modules/token/components/burn-token/open-api.ts (hunk 3 above)
Comment (rex-god) [reply src/modules/token/components/burn-token/open-api.ts:15]: Yes

File: src/modules/token/components/burn-token/burn-token.service.ts (hunk 2 above)
Comment (rex-god) [reply src/modules/token/components/burn-token/burn-token.service.ts:48]: Ok

File: src/modules/token/components/burn-token/burn-token.controller.ts (hunk 1 above)
Comment (rex-god) [reply src/modules/token/components/burn-token/burn-token.controller.ts:16]: Due to copying. I'll fix it.

File: src/modules/token/components/create-token/create-token.service.ts (hunk 4 above)
Comment (rex-god) [reply src/modules/token/components/create-token/create-token.service.ts:103]: What are you talking about? 🤔

**Pull request closed without merging**